### PR TITLE
1892 Fix E2E test

### PR DESCRIPTION
**Pull request by leite08**
Ticket: https://github.com/metriport/metriport-internal/issues/1892

### Dependencies

- Upstream: <link to PR that introduced the issue>
- Downstream: 

### Description

This fixes E2E tests that broke after we introduced some changes on how we build the consolidated payload on the API - [context](https://metriport.slack.com/archives/C04FZ9859FZ/p1718990218540019).

- `PUT /consolidated` now uses the req ID to identify diff payloads stored on S3
   - and it returns the req ID so we can use it on E2E tests
- concentrate header names in the `header.ts` file
- create a `getRequestId()` function that uses the local storage

### Testing

WIP

- Local
  - [x] unit and E2E tests pass
  - [ ] ...
- Staging
  - [ ] ...
- Sandbox
  - [ ] ...
- Production
  - [ ] ...

### Release Plan

- [ ] Merge this

### CODE REVIEW

**Comment by RamilGaripov** [comment]
Approving @leite08 's commits

**Review by RamilGaripov — APPROVED** [review]
Approving @leite08 's commit

File: packages/api/src/external/fhir/shared/extensions/extension.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { BASE_EXTENSION_URL } from "./base-extension";
 
 // TODO #712: create this extension
 export const DATA_SOURCE_EXTENSION_URL = `${BASE_EXTENSION_URL}/data-source.json`;
+export const DOC_ID_EXTENSION_URL = `${BASE_EXTENSION_URL}/doc-id-extension.json`;
```
Comment (leite08) [inline packages/api/src/external/fhir/shared/extensions/extension.ts:5]: This already exists on the `doc-id-extension.ts` file, let's remove from there so its set in only one place?

File: packages/api/src/external/fhir/shared/hydrate-bundle.ts [hunk 2]
```diff
@@ -14,7 +15,7 @@ export function hydrateBundle(
   fhirBundleDestinationKey: string
 ): ValidBundle {
   const docExtension: Extension = {
-    url: "https://public.metriport.com/fhir/StructureDefinition/doc-id-extension.json",
+    url: DOC_ID_EXTENSION_URL,
     valueString: fhirBundleDestinationKey,
   };
```
Comment (leite08) [inline packages/api/src/external/fhir/shared/hydrate-bundle.ts:20]: There's a function to build this - `buildDocIdFHIRExtension()`, let's use it?

Comment (leite08) [reply packages/api/src/external/fhir/shared/hydrate-bundle.ts:20]: BTW, that fn doesn't set the `valueCoding` of the extension - while the `dataSourceExtensionDefaults` does. Shouldn't it?

File: packages/api/src/external/fhir/shared/extensions/extension.ts (hunk 1 above)
Comment (RamilGaripov) [reply packages/api/src/external/fhir/shared/extensions/extension.ts:5]: Oh, good catch. I didn't realize we had duplicate code there in API vs Core. I'll deprecate the ones on the API

File: packages/api/src/external/fhir/shared/hydrate-bundle.ts (hunk 2 above)
Comment (RamilGaripov) [reply packages/api/src/external/fhir/shared/hydrate-bundle.ts:20]: No, i think it's fine that it doesn't use `valueCoding`. Extensions are allowed to have values of different types, and we have this extension referenced in this format in a bunch of places

File: packages/api/src/external/fhir/shared/extensions/extension.ts
Comment (leite08) [inline packages/api/src/external/fhir/shared/extensions/extension.ts:1]: 💅

**Comment by leite08** [comment]
Approving @RamilGaripov 's commits

**Pull request merged**